sklearn/svm/tests/test_svm.py

```diff
@@ -66,12 +66,54 @@ def test_libsvm_iris():
     assert_array_equal(clf.classes_, np.sort(clf.classes_))
 
     # check also the low-level API
-    model = _libsvm.fit(iris.data, iris.target.astype(np.float64))
-    pred = _libsvm.predict(iris.data, *model)
+    (
+        libsvm_support,
+        libsvm_support_vectors,
+        libsvm_n_class_SV,
+        libsvm_sv_coef,
+        libsvm_intercept,
+        libsvm_probA,
+        libsvm_probB,
+        libsvm_fit_status,
+        libsvm_num_iter,
+    ) = _libsvm.fit(iris.data, iris.target.astype(np.float64))
+
+    lib_svm_model = (
+        libsvm_support,
+        libsvm_support_vectors,
+        libsvm_n_class_SV,
+        libsvm_sv_coef,
+        libsvm_intercept,
+        libsvm_num_iter,
+        libsvm_probA,
+        libsvm_probB,
+    )
+    pred = _libsvm.predict(iris.data, *lib_svm_model)
```

Comment:
While coding this, I realized that this will improve code readability for the test, but I will worsen the code for the `predict`, `predict_proba`, `decision_function`, `cross_validation` functions from the Cython code, and all the functions from the `BaseLibSVM` class.

Since we will add a new class as the return type of the fit function, we will have to copy every attribute of `FittedLibSVM` in the `BaseLibSVM` class inside the fit methods (`_dense_fit` and `_sparse_fit`). We can define a method that does this.

Also, for every method of `BaseLibSVM` that depends on the fitted model (`predict`, `predict_proba`, `decision_function`), we will need to create a new instance of `FittedLibSVM` and pass it to the corresponding method. This could also be a new method that creates an instance of `FittedLibSVM` from self.

I wonder if it is not better to leave it like it is now with the unpacked values in the tests. What do you think?

I will try to add the changes so you can see what I mean. Maybe I'm over-complicating things that are simple.

Reply:
Done, at least for the dense case.
Overall, we added more lines of code with this approach, but I think it is clear.

@jjerphan, Let me know what you think.
Any comments or suggestions are more than welcome.

If it is ok, I will improve the documentation and implement the same extension type for the sparse case.

Reply:
> While coding this, I realized that this will improve code readability for the test, but I will worsen the code for the predict, predict_proba, decision_function, cross_validation functions from the Cython code, and all the functions from the BaseLibSVM class.

I think you are right.

a44b4a4 add is valid but creates too much changes and complexifies the implementation (I thought it would have been simpler but I was wrong).

I agree with you and think having just the unpacking for this test is better.

In this case a44b4a4 can be reverted and a comment can be added explain why the unpacking and packing are done. For instance (I can't suggest):

```python
    # We unpack the be able to repack some of the
    # return values from libsvm.
    (
        libsvm_support,
        libsvm_support_vectors,
        libsvm_n_class_SV,
        libsvm_sv_coef,
        libsvm_intercept,
        libsvm_probA,
        libsvm_probB,
        # libsvm_fit_status won't be packed bellow.
        libsvm_fit_status,
        libsvm_num_iter,
    ) = _libsvm.fit(iris.data, iris.target.astype(np.float64))

    lib_svm_model = (
        libsvm_support,
        libsvm_support_vectors,
        libsvm_n_class_SV,
        libsvm_sv_coef,
        libsvm_intercept,
        libsvm_num_iter,
        libsvm_probA,
        libsvm_probB,
    )
    pred = _libsvm.predict(iris.data, *lib_svm_model)
```

What do you think?

Reply:
Done ^^
Btw, congrats @jjerphan on your new role in the project :nerd_face:

Reply:
Thank you, @jmloyola.

---
"We do what we must because we can."